openedx/adg/common/mailchimp_pipeline/client.py

```diff
@@ -0,0 +1,124 @@
+"""
+Mailchimp Connection Client
+This code is used to sync users with mailchimp subscriber list
+"""
+import hashlib
+import json
+
+from django.conf import settings
+from requests import request
+from requests.exceptions import HTTPError
+
+
+class MailChimpException(Exception):
+    pass
+
+
+class Connection(object):
+    """ mailchimp api connection """
+
+    output = "json"
+    version = '3.0'
+
+    def __init__(self, apikey=None, secure=False):
+        self._apikey = apikey
+
+        proto = 'https' if secure else 'http'
+        dc = apikey.split('-')[1]
+
+        self.root = '{}://{}.api.mailchimp.com/{}/'.format(proto, dc, self.version)
+
+    def make_request(self, method="GET", path=None, **kwargs):
+        """
+        This function makes the desired request to mailchimp and performs actions on the subscriber list
+        """
```

Comment:
We use Google style docstring. For reference, https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html